windows/ReactTestApp/MainPage.cpp

```diff
@@ -43,12 +43,14 @@ namespace winrt::ReactTestApp::implementation
                 menuItems.Append(newMenuItem);
             }
 
+            ReactRootView().ReactNativeHost(reactInstance_.ReactHost());
+
             // If only one component is present load it automatically
             if (components.size() == 1) {
-                ReactRootView().ComponentName(to_hstring(components.at(0).appKey));
+                auto firstComponent = components.at(0);
+                ReactRootView().ComponentName(to_hstring(firstComponent.appKey));
+                WriteInitialProperties(firstComponent.initialProperties);
```

Comment:
If we can move the `ReloadInstance()` call to `MenuFlyoutItem` click handler, we can reuse `LoadReactComponent` here:
```suggestion
                auto firstComponent = components.at(0);
                LoadReactComponent(firstComponent.appKey, firstComponent.initialProperties);
```